plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlValidator.java

```diff
@@ -164,11 +164,28 @@ private void validateJsp() {
           Node jspNode = jspList.item(i);
           String jspName = jspNode.getTextContent();
           IFile file = root.getFile(jspName).getUnderlyingFile();
-          if (!file.exists()) {
-            DocumentLocation location = (DocumentLocation) jspNode.getUserData("location");
-            BannedElement element = new JspFileElement(jspName, location, jspName.length());
-            blacklist.add(element);
+          // Looks in src/main/webapp
+          if (file.exists()) {
+            continue;
           }
+          file = root.getFile("WEB-INF/" + jspName).getUnderlyingFile();
+          // Looks in src/main/webapp/WEB-INF
+          if (file.exists()) {
+            continue;
+          }
+          file = root.getFile("WEB-INF/classes/" + jspName).getUnderlyingFile();
+          // Looks in src/main/java
+          if (file.exists()) {
+            continue;
+          }
+          file = root.getFile("WEB-INF/lib/" + jspName).getUnderlyingFile();
```

Comment:
This isn't quite right: `lib` will contain jars.  It looks like any [JSPs must be in `META-INF/resources`](http://stackoverflow.com/a/11063065/600339), though I'm not sure if this works for Servlet 2.5.